scripts/esmDependencyCheck.js

```diff
@@ -0,0 +1,161 @@
+  Object.keys(timeJson).forEach(version => {
+    const splitVersionArray = version.split('.');
+    const ourRegex = new RegExp(`^${versionMatcherString}`, 'gi');
+    if (version.match(ourRegex)) {
+      versionToReturn = version;
+    }
+  });
+  return versionToReturn;
+};
+
+const getRepoFromNpmData = npmData => {
+  if (
+    npmData.hasOwnProperty('repository') &&
+    npmData.repository.hasOwnProperty('url')
+  ) {
+    return npmData.repository.url;
+  } else if (npmData.hasOwnProperty('repository')) {
+    return npmData.repository;
+  }
+  return npmData.url || npmData.homepage;
+};
+
+Object.keys(allDependencies).forEach((dep, i) => {
+  if (gitToken) {
+    let gitRepo;
+    const cmd = exec(`npm view ${dep} --json`, (err, stdout, stderr) => {
+      if (err) {
+        console.error(err);
+      }
+      if (stdout) {
+        const depRepository = JSON.parse(stdout);
+        const modifiedDate = new Date(depRepository.time.modified);
+
+        const ourVersion = dealWithCaretsAndTildes(
+          allDependencies[dep],
+          stdout,
+        );
+
+        const dateOfOurVersion = new Date(depRepository.time[ourVersion]);
+
+        const ourFreshnessInDays = datediff(
+          dateOfOurVersion.getTime(),
+          dateNow,
+        );
+
+        gitRepo = getRepoFromNpmData(depRepository);
+        if (gitRepo) {
+          gitRepo = gitRepo.replace(
+            /^(?:git)?(?:\+)?(https|ssh)?:\/\/(?:git@)?github.com\/|\.git(?:#?.*)$/g,
+            '',
+          );
+          const remoteGitFile = getRemoteGitFile(gitRepo).then(
+            gitDepModuleType => {
+              dependencyTable.push({
+                name: dep,
+                type: gitDepModuleType,
+                mostRecentVersion: modifiedDate,
+                ourVersion: dateOfOurVersion,
+                ourFreshnessInDays,
+              });
+            },
+          );
+        } else {
+          console.error(
+            `dep ${dep} has no public repo so we're reading from local`,
+          );
+          const depRepository = JSON.parse(
+            fs.readFileSync(`./node_modules/${dep}/package.json`),
+          );
+          const depType = depRepository.type ? depRepository.type : 'commonjs';
+          dependencyTable.push({
+            name: dep,
+            type: depType,
+            mostRecentVersion: modifiedDate,
+            ourVersion: dateOfOurVersion,
+            ourFreshnessInDays,
+          });
+        }
+      } else {
+        console.log('no stdout', dep, stdout);
+      }
+
+      let csvContents =
+        'Dependency,Type, Most Recent Version, Our Version, Our Version Freshness in Days';
+      dependencyTable.forEach(
+        ({ name, type, mostRecentVersion, ourVersion, ourFreshnessInDays }) => {
+          csvContents += `\n${name},${type},${mostRecentVersion},${ourVersion},${ourFreshnessInDays}`;
+        },
+      );
+      fs.writeFileSync('./esmDependencyTable.csv', csvContents);
```

Comment:
why doesn't the csv in this PR reflect the headers? Seems to be the dep freshness one (with downloads in it)

Reply:
eerrrr weird, will look into this. Wonder if I committed a version from earlier before I finished writing the script. I didn't push the new csv

Reply:
I have deleted it. When I include one (want to check the accuracy first) then a new one will be pushed.